StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/BottomSheetViewController.swift

```diff
@@ -281,8 +281,8 @@ extension BottomSheetViewController: PaymentSheetAuthenticationContext {
         completion()
     }
 
-    func presentPollingVCForAction(_ action: STPPaymentHandlerActionParams) {
-        let pollingVC = PollingViewController(currentAction: action,
+    func presentPollingVCForAction(action: STPPaymentHandlerActionParams, type: STPPaymentMethodType) {
+        let pollingVC = PollingViewController(currentAction: action, type: type,
```

Comment:
Based on our zoom call lets create a new view model for the PollingViewController which computes the CTA and deadline, then inject it into the PollingViewController here, and also write tests for the new view controller.

The view model should compute the CTA and deadline based on the payment method type, e.g. inject the payment method type into the view model.